### PR TITLE
Avoid time.sleep in TestRecentSequenceHistory

### LINKED ISSUE #3004: Sporadic test failure: TestRecentSequenceHistory
Originally filed in https://github.com/couchbase/sync_gateway/issues/1589

### PR DESCRIPTION
**Pull request by adamcfraser**
Also removes hardcoded test expectations for value of kMaxRecentSequences

Fixes #3004

### CODE REVIEW

File: db/database_test.go [hunk 1]
```diff
@@ -1240,34 +1240,37 @@ func TestRecentSequenceHistory(t *testing.T) {
...
+	assert.DeepEquals(t, doc.RecentSequences, expectedRecent)
 
-	// Add many sequences to validate pruning when past max (20)
-	for i := 0; i < kMaxRecentSequences; i++ {
-		revid, err = db.Put("doc1", body)
-		// Sleep needed to ensure consistent results when running single-threaded vs. multi-threaded test:
-		// without it we can't predict the relative update times of nextSequence and RecentSequences
-		time.Sleep(50 * time.Millisecond)
-	}
+	// Wait for db.sequences.nextSequence to catch up (used by pruning calculation)
```
Comment (tleyden) [inline db/database_test.go]: I don't fully understand this comment .. can you add a bit more explaining why it needs to catch up?

Comment (tleyden) [reply db/database_test.go]: Would this also work?  `db.changeCache.waitForSequence(seqTracker)`?

Comment (tleyden) [reply db/database_test.go]: > // Wait for db.sequences.nextSequence to catch up (used by pruning calculation)

Oh, so basically the change cache needs to catch up on the async dcp feed?

Comment (adamcfraser) [reply db/database_test.go]: Good point - should use seqTracker here.  I'll also add more details to the comment.

**Pull request merged**